### PR TITLE
[SDL-0266] New vehicle data GearStatus

### LINKED ISSUE #1522: [SDL 0266] New vehicle data GearStatus
Proposal: [New vehicle data GearStatus](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0266-New-vehicle-data-GearStatus.md)

> This proposal is to add new vehicle data `GearStatus` to know drive mode and actual gear the transmission has selected.

Review: https://github.com/smartdevicelink/sdl_evolution/issues/881

Steering Committee Decision:

>The Steering Committee voted to accept this proposal with the following revisions: use `userSelectedGear` in place of `selectedGear` and use the descriptions  for `GearStatus` included in [this comment](https://github.com/smartdevicelink/sdl_evolution/issues/881#issuecomment-573838656).
> The proposal .md file was updated to reflect these revisions on 1/21/2020.

### PR DESCRIPTION
**Pull request by yoooriii**
Fixes #1522   -- New vehicle data GearStatus

This PR is **ready** for review.

### Risk
This PR makes **minor** API changes.

### Testing Plan
- [x] I have verified that I have not introduced new warnings in this PR
- [x] I have run the unit tests with this PR

#### Unit Tests
Additional Test cases implemented and ran on my local machine under Xcode and in Terminal on the iPhone-8 simulator running iOS-13.0
Among all tests only the 5 tests connected with SDLLockScreenViewControllerSnapshotTests fail though I do not think it is new or in any way connected with this update.

### Summary
[proposal in sdl evolution](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0266-New-vehicle-data-GearStatus.md)

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

**Comment by yoooriii** [comment]
@atiwari9 could you please review it?

File: SmartDeviceLink-iOS.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -1543,6 +1543,12 @@
...
+		B38D8E6D24A034DB00B977D0 /* SDLGearStatus.h in Headers */ = {isa = PBXBuildFile; fileRef = B38D8E6B24A034DB00B977D0 /* SDLGearStatus.h */; settings = {ATTRIBUTES = (Public, ); }; };
+		B38D8E6E24A034DB00B977D0 /* SDLGearStatus.m in Sources */ = {isa = PBXBuildFile; fileRef = B38D8E6C24A034DB00B977D0 /* SDLGearStatus.m */; };
+		B38D8E7124A0351200B977D0 /* SDLTransmissionType.m in Sources */ = {isa = PBXBuildFile; fileRef = B38D8E6F24A0351200B977D0 /* SDLTransmissionType.m */; };
+		B38D8E7224A0351200B977D0 /* SDLTransmissionType.h in Headers */ = {isa = PBXBuildFile; fileRef = B38D8E7024A0351200B977D0 /* SDLTransmissionType.h */; settings = {ATTRIBUTES = (Public, ); }; };
+		B38D8E7E24A118BD00B977D0 /* SDLGearStatusSpec.m in Sources */ = {isa = PBXBuildFile; fileRef = B38D8E7D24A118BD00B977D0 /* SDLGearStatusSpec.m */; };
+		B38D8E8024A1E3D000B977D0 /* SDLTransmissionTypeSpec.m in Sources */ = {isa = PBXBuildFile; fileRef = B38D8E7F24A1E3D000B977D0 /* SDLTransmissionTypeSpec.m */; };
```
Comment (SatbirTanda) [inline SmartDeviceLink-iOS.xcodeproj/project.pbxproj:1551]: Add new files to the `podspec` files

File: SmartDeviceLink/SDLRPCParameterNames.h [hunk 2]
```diff
@@ -724,5 +754,9 @@ extern SDLRPCParameterName const SDLRPCParameterNameWindowTypeSupported;
 extern SDLRPCParameterName const SDLRPCParameterNameX;
 extern SDLRPCParameterName const SDLRPCParameterNameY;
 extern SDLRPCParameterName const SDLRPCParameterNameYear;
+extern SDLRPCParameterName const SDLRPCParameterNameActualGear;
+extern SDLRPCParameterName const SDLRPCParameterNameGearStatus;
+extern SDLRPCParameterName const SDLRPCParameterNameUserSelectedGear;
+extern SDLRPCParameterName const SDLRPCParameterNameTransmissionType;
```
Comment (SatbirTanda) [inline SmartDeviceLink/SDLRPCParameterNames.h:760]: Should be added alphabetically

File: SmartDeviceLink/SDLRPCParameterNames.m [hunk 3]
```diff
@@ -719,5 +749,9 @@
 SDLRPCParameterName const SDLRPCParameterNameX = @"x";
 SDLRPCParameterName const SDLRPCParameterNameY = @"y";
 SDLRPCParameterName const SDLRPCParameterNameYear = @"year";
+SDLRPCParameterName const SDLRPCParameterNameActualGear = @"actualGear";
+SDLRPCParameterName const SDLRPCParameterNameGearStatus = @"gearStatus";
+SDLRPCParameterName const SDLRPCParameterNameUserSelectedGear = @"userSelectedGear";
+SDLRPCParameterName const SDLRPCParameterNameTransmissionType = @"transmissionType";
```
Comment (SatbirTanda) [inline SmartDeviceLink/SDLRPCParameterNames.m:755]: alphabetically

File: SmartDeviceLinkTests/RPCSpecs/NotificationSpecs/SDLOnVehicleDataSpec.m [hunk 4]
```diff
@@ -44,6 +46,7 @@
         myKey = [[SDLMyKey alloc] init];
         fuelRange = [[SDLFuelRange alloc] init];
         cloudAppVehicleID = @"testCloudAppVehicleID";
+        gearStatus = [SDLGearStatus new];
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/NotificationSpecs/SDLOnVehicleDataSpec.m:49]: `alloc init`

File: SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLGetVehicleDataSpec.m [hunk 5]
```diff
@@ -16,7 +16,7 @@
 
 describe(@"Getter/Setter Tests", ^ {
     it(@"Should set and get correctly", ^ {
-        SDLGetVehicleData* testRequest = [[SDLGetVehicleData alloc] init];
+        SDLGetVehicleData* testRequest = [SDLGetVehicleData new];
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLGetVehicleDataSpec.m:19]: `alloc init`

File: SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLGetVehicleDataSpec.m [hunk 6]
```diff
@@ -151,7 +165,7 @@
 
 describe(@"initializers", ^{
     context(@"init", ^{
-        SDLGetVehicleData* testRequest = [[SDLGetVehicleData alloc] init];
+        SDLGetVehicleData* testRequest = [SDLGetVehicleData new];
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLGetVehicleDataSpec.m:168]: `alloc init`

File: SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLSubscribeVehicleDataSpec.m [hunk 7]
```diff
@@ -16,7 +16,7 @@
 
 describe(@"Getter/Setter Tests", ^ {
     it(@"Should set and get correctly", ^ {
-        SDLSubscribeVehicleData* testRequest = [[SDLSubscribeVehicleData alloc] init];
+        SDLSubscribeVehicleData* testRequest = [SDLSubscribeVehicleData new];
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLSubscribeVehicleDataSpec.m:19]: `alloc init`

File: SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLSubscribeVehicleDataSpec.m [hunk 8]
```diff
@@ -151,7 +165,7 @@
 
 describe(@"initializers", ^{
     context(@"init", ^{
-        SDLSubscribeVehicleData* testRequest = [[SDLSubscribeVehicleData alloc] init];
+        SDLSubscribeVehicleData* testRequest = [SDLSubscribeVehicleData new];
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLSubscribeVehicleDataSpec.m:168]: `alloc init`

File: SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLUnsubscribeVehicleDataSpec.m [hunk 9]
```diff
@@ -16,7 +16,7 @@
 
 describe(@"Getter/Setter Tests", ^ {
     it(@"Should set and get correctly", ^ {
-        SDLUnsubscribeVehicleData* testRequest = [[SDLUnsubscribeVehicleData alloc] init];
+        SDLUnsubscribeVehicleData* testRequest = [SDLUnsubscribeVehicleData new];
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLUnsubscribeVehicleDataSpec.m:19]: `alloc init`

File: SmartDeviceLinkTests/RPCSpecs/ResponseSpecs/SDLGetVehicleDataResponseSpec.m [hunk 10]
```diff
@@ -46,10 +48,11 @@
         fuelRange = [[SDLFuelRange alloc] init];
         vin = @"6574839201a";
         cloudAppVehicleID = @"cloudAppVehicleID";
+        gearStatus = [SDLGearStatus new];
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/ResponseSpecs/SDLGetVehicleDataResponseSpec.m:51]: `alloc init`

File: SmartDeviceLinkTests/RPCSpecs/ResponseSpecs/SDLGetVehicleDataResponseSpec.m [hunk 11]
```diff
@@ -46,10 +48,11 @@
...
     });
 
     it(@"Should set and get correctly", ^ {
-        SDLGetVehicleDataResponse* testResponse = [[SDLGetVehicleDataResponse alloc] init];
+        SDLGetVehicleDataResponse* testResponse = [SDLGetVehicleDataResponse new];
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/ResponseSpecs/SDLGetVehicleDataResponseSpec.m:55]: `alloc init`

File: SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLGearStatusSpec.m [hunk 12]
```diff
@@ -0,0 +1,45 @@
...
+        SDLGearStatus* testStruct = [[SDLGearStatus alloc] initWithDictionary:dict];
+#pragma clang diagnostic pop
+
+        expect(testStruct.userSelectedGear).to(equal(SDLPRNDLNinth));
+        expect(testStruct.actualGear).to(equal(SDLPRNDLTenth));
+        expect(testStruct.transmissionType).to(equal(SDLTransmissionTypeAutomatic));
+    });
+
+    it(@"Should return nil if not set", ^ {
+        SDLGearStatus* testStruct = [SDLGearStatus new];
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/StructSpecs/SDLGearStatusSpec.m:37]: `alloc init`

File: SmartDeviceLinkTests/RPCSpecs/ResponseSpecs/SDLSubscribeVehicleDataResponseSpec.m [hunk 13]
```diff
@@ -117,7 +119,9 @@
                                                                    SDLRPCParameterNameSteeringWheelAngle:vehicleDataResult,
                                                                    SDLRPCParameterNameTirePressure:vehicleDataResult,
                                                                    SDLRPCParameterNameTurnSignal:vehicleDataResult,
-                                                                   SDLRPCParameterNameWiperStatus:vehicleDataResult},
+                                                                   SDLRPCParameterNameWiperStatus:vehicleDataResult,
+                                                                   @"gearStatus":vehicleDataResult
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/ResponseSpecs/SDLSubscribeVehicleDataResponseSpec.m:123]: `SDLRPCParameterNameGearStatus`

File: SmartDeviceLinkTests/RPCSpecs/ResponseSpecs/SDLGetVehicleDataResponseSpec.m [hunk 14]
```diff
@@ -147,8 +158,10 @@
                                                      SDLRPCParameterNameTirePressure:tires,
                                                      SDLRPCParameterNameTurnSignal:SDLTurnSignalOff,
                                                      SDLRPCParameterNameVIN:vin,
-                                                     SDLRPCParameterNameWiperStatus:SDLWiperStatusAutomaticHigh},
-                                             SDLRPCParameterNameOperationName:SDLRPCFunctionNameGetVehicleData}} mutableCopy];
+                                                     SDLRPCParameterNameWiperStatus:SDLWiperStatusAutomaticHigh,
+                                                     @"gearStatus":gearStatus
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/ResponseSpecs/SDLGetVehicleDataResponseSpec.m:162]: Use `SDLRPCParameterNameGearStatus`

File: SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLUnsubscribeVehicleDataSpec.m [hunk 15]
```diff
@@ -110,7 +118,9 @@
                                                            SDLRPCParameterNameSteeringWheelAngle:@YES,
                                                            SDLRPCParameterNameTirePressure:@YES,
                                                            SDLRPCParameterNameTurnSignal:@YES,
-                                                           SDLRPCParameterNameWiperStatus:@YES},
+                                                           SDLRPCParameterNameWiperStatus:@YES,
+                                                           @"gearStatus":@YES
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLUnsubscribeVehicleDataSpec.m:122]: Use `SDLRPCParameterNameGearStatus`

File: SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLSubscribeVehicleDataSpec.m [hunk 16]
```diff
@@ -110,7 +118,9 @@
                                                            SDLRPCParameterNameSteeringWheelAngle:@NO,
                                                            SDLRPCParameterNameTirePressure:@YES,
                                                            SDLRPCParameterNameTurnSignal:@NO,
-                                                           SDLRPCParameterNameWiperStatus:@NO},
+                                                           SDLRPCParameterNameWiperStatus:@NO,
+                                                           @"gearStatus":@YES
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLSubscribeVehicleDataSpec.m:122]: `SDLRPCParameterNameGearStatus`

File: SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLGetVehicleDataSpec.m [hunk 17]
```diff
@@ -110,7 +118,9 @@
                                                                    SDLRPCParameterNameSteeringWheelAngle:@NO,
                                                                    SDLRPCParameterNameTirePressure:@YES,
                                                                    SDLRPCParameterNameTurnSignal:@NO,
-                                                                   SDLRPCParameterNameWiperStatus:@YES},
+                                                                   SDLRPCParameterNameWiperStatus:@YES,
+                                                                   @"gearStatus":@YES
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/RequestSpecs/SDLGetVehicleDataSpec.m:122]: `SDLRPCParameterNameGearStatus`

File: SmartDeviceLinkTests/RPCSpecs/NotificationSpecs/SDLOnVehicleDataSpec.m [hunk 18]
```diff
@@ -138,6 +143,7 @@
                                            SDLRPCParameterNameMyKey:myKey,
                                            SDLRPCParameterNameOdometer:@100050,
                                            SDLRPCParameterNamePRNDL:SDLPRNDLDrive,
+                                           @"gearStatus":gearStatus,
```
Comment (SatbirTanda) [inline SmartDeviceLinkTests/RPCSpecs/NotificationSpecs/SDLOnVehicleDataSpec.m:146]: `SDLRPCParameterNameGearStatus`

**Comment by yoooriii** [comment]
Hello, @theresalech could you please review?

**Comment by theresalech** [comment]
@yoooriii please make sure the feedback provided in https://github.com/smartdevicelink/sdl_ios/pull/1693#pullrequestreview-450716620 is reflected in this PR as well, and let us know once this has been confirmed/ready for review.  Thank you!

**Comment by yoooriii** [comment]
Hello @theresalech The suggested changes #1693 (review) is applied in this PR. Thank you!

**Comment by joeljfischer** [comment]
@yoooriii, we will not be reviewing this PR again until [this PR](https://github.com/smartdevicelink/sdl_ios/pull/1693) is fully reviewed and merged in order to reduce the amount of time that we are spending in giving the same review comments and so that you have an approved PR to know exactly what we're looking for. We ask that you place your focus on that PR.

**Comment by yoooriii** [comment]
Hello @NicoleYarroch, could you please review the latest changes?

**Comment by yoooriii** [comment]
Generally it is ready though I need to resolve merge conflicts.

**Comment by NicoleYarroch** [comment]
@yoooriii I see that there are now 264 files changed in this PR which makes me think something went wrong during your merge of develop. Please update this PR so it only reflects changes necessary for this PR as a lot of non-relevant code has been changed.

**Comment by NicoleYarroch** [comment]
@yoooriii Please tag me once this PR is ready for review.

**Comment by yoooriii** [comment]
Hello @NicoleYarroch @joeljfischer , could you please review? Note: having merged with develop I got 264 files updated so I had no other choice but to force push another merge commit.

File: SmartDeviceLink/SDLPRNDL.h [hunk 19]
```diff
@@ -46,11 +46,21 @@ extern SDLPRNDL const SDLPRNDLLowGear;
...
+ * Ninth gear
+ */
+extern SDLPRNDL const SDLPRNDLNinth;
+
 /**
  * Second gear
  */
 extern SDLPRNDL const SDLPRNDLSecond;
 
+/**
```
Comment (yoooriii) [inline SmartDeviceLink/SDLPRNDL.h:59]: Ninth--Second--Tenth : sorted in alpha order

File: SmartDeviceLink/SDLPRNDL.h [hunk 20]
```diff
@@ -46,11 +46,21 @@ extern SDLPRNDL const SDLPRNDLLowGear;
...
+ * Tenth gear
+ */
+extern SDLPRNDL const SDLPRNDLTenth;
+
```
Comment (yoooriii) [inline SmartDeviceLink/SDLPRNDL.h:63]: Ninth--Second--Tenth : sorted in alpha order.

**Comment by NicoleYarroch** [comment]
@yoooriii Hi, I was just checking in to see if you have an estimated date for when the requested changes will be done? Let me know if you have any questions. Thanks.

**Comment by yoooriii** [comment]
Hello @NicoleYarroch, the PR updated, could you please review? Thank you.

**Comment by NicoleYarroch** [comment]
@yoooriii I left a request for one change.

**Comment by yoooriii** [comment]
Hello @NicoleYarroch, I have updated SDLVehicleDataTypePRNDL, could you please review? Thank you.

**Pull request merged**